# Fix crash when host has not defined shared directory
# channel-webdav.c:369 webdav-11:0: starting client 1814016624 Segmentation fault 

--- a/src/channel-webdav.c
+++ b/src/channel-webdav.c
@@ -364,42 +364,46 @@
     bool started;
 
     session = spice_channel_get_session(SPICE_CHANNEL(self));
-    server = phodav_server_get_soup_server(spice_session_get_webdav_server(session));
 
-    CHANNEL_DEBUG(self, "starting client %" G_GINT64_FORMAT, c->demux.client);
+  	 if(spice_session_get_shared_dir(session)!= NULL) {
 
-    client = g_new0(Client, 1);
-    client->refs = 1;
-    client->id = c->demux.client;
-    client->self = self;
-    client->mux.id = GINT64_TO_LE(client->id);
-    client->mux.buf = g_malloc0(MAX_MUX_SIZE);
-    client->cancellable = g_cancellable_new();
-    spice_make_pipe(&client->pipe, &peer);
-
-G_GNUC_BEGIN_IGNORE_DEPRECATIONS
-    addr = g_inet_socket_address_new_from_string ("127.0.0.1", 0);
-G_GNUC_END_IGNORE_DEPRECATIONS
-    if (!soup_server_accept_iostream(server, peer, addr, addr, &error))
-        goto fail;
-
-    g_hash_table_insert(c->clients, &client->id, client);
-
-    started = client_start_read(client);
-    g_assert(started);
-    demux_to_client(client);
-
-    g_clear_object(&addr);
-    return;
-
-fail:
-    if (error)
-        CHANNEL_DEBUG(self, "failed to start client: %s", error->message);
-
-    g_clear_object(&addr);
-    g_clear_object(&peer);
-    g_clear_error(&error);
-    client_unref(client);
+	    server = phodav_server_get_soup_server(spice_session_get_webdav_server(session));
+
+   	 CHANNEL_DEBUG(self, "starting client %" G_GINT64_FORMAT, c->demux.client);
+
+   	client = g_new0(Client, 1);
+    	client->refs = 1;
+    	client->id = c->demux.client;
+    	client->self = self;
+    	client->mux.id = GINT64_TO_LE(client->id);
+    	client->mux.buf = g_malloc0(MAX_MUX_SIZE);
+    	client->cancellable = g_cancellable_new();
+    	spice_make_pipe(&client->pipe, &peer);
+
+		G_GNUC_BEGIN_IGNORE_DEPRECATIONS
+    	addr = g_inet_socket_address_new_from_string ("127.0.0.1", 0);
+		G_GNUC_END_IGNORE_DEPRECATIONS
+    	if (!soup_server_accept_iostream(server, peer, addr, addr, &error))
+      	  goto fail;
+
+   	 g_hash_table_insert(c->clients, &client->id, client);
+
+    	started = client_start_read(client);
+    	g_assert(started);
+	   demux_to_client(client);
+
+   	g_clear_object(&addr);
+    	return;
+
+	fail:
+   	 if (error)
+       		 CHANNEL_DEBUG(self, "failed to start client: %s", error->message);
+
+   	 g_clear_object(&addr);
+  	 	 g_clear_object(&peer);
+   	 g_clear_error(&error);
+   	 client_unref(client);
+	}
 #endif
 }

